pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -102,6 +104,14 @@ func (configgen *ConfigGeneratorImpl) BuildListeners(env model.Environment, node
 	return nil, nil
 }
 
+// sortServicesByCreationTime sorts the list of services in ascending order by their creation time (if available).
+func sortServicesByCreationTime(services []*model.Service) []*model.Service {
+	sort.SliceStable(services, func(i, j int) bool {
+		return services[i].CreationTime.Before(services[j].CreationTime)
```

Comment:
what happens if time is not set? its all 0s ?

Reply:
I'm using a stable sort, so the original order should be maintained.